Revert/Sources/KeyboardHandler.swift

```diff
@@ -17,44 +17,63 @@ final class KeyboardHandler: NSObject {
+  
+  func keyboardWillShowHideNotification()
+}
 
-    UIView.animateWithDuration(animationDuration, delay: 0, options: animationOptions, animations: {
-      scrollView.contentInset = contentInsets
-      scrollView.scrollIndicatorInsets = scrollIndicatorInsets
-      }, completion: nil)
+extension KeyboardHandlerProtocol {
+  func registerNotifications() {
+    // This empty method simulates an optional protocol method like we have in ObjC.
   }
-
-  // MARK: Private
-
-  private func registerNotifications() {
-    NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(self.keyboardWillShowHideNotification(_:)), name: UIKeyboardWillShowNotification, object: nil)
-    NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(self.keyboardWillShowHideNotification(_:)), name: UIKeyboardWillHideNotification, object: nil)
+  
+  func unregisterNotifications() {
+    // This empty method simulates an optional protocol method like we have in ObjC.
   }
-
-  private func unregisterNotifications() {
-    NSNotificationCenter.defaultCenter().removeObserver(self, name: UIKeyboardWillShowNotification, object: nil)
-    NSNotificationCenter.defaultCenter().removeObserver(self, name: UIKeyboardWillHideNotification, object: nil)
+  
+  func keyboardWillShowHideNotification() {
+    // This empty method simulates an optional protocol method like we have in ObjC.
   }
 }
+
+#if os(iOS)
+  extension KeyboardHandler {
+    func keyboardWillShowHideNotification(notification: NSNotification) {
+      guard let scrollView = self.scrollView,
+        scrollViewSuperview = scrollView.superview,
+        viewController = self.viewController,
+        animationDuration = notification.userInfo?[UIKeyboardAnimationDurationUserInfoKey] as? NSTimeInterval,
+        animationCurveInt = notification.userInfo?[UIKeyboardAnimationCurveUserInfoKey] as? Int,
+        animationCurve = UIViewAnimationCurve(rawValue: animationCurveInt),
+        keyboardFrameValue = notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue else {
+          return
+      }
+      
+      let keyboardFrame = keyboardFrameValue.CGRectValue()
+      let keyboardFrameInView = scrollViewSuperview.convertRect(keyboardFrame, fromView: nil)
+      let bottomInset = max(scrollView.frame.maxY - keyboardFrameInView.minY, viewController.bottomLayoutGuide.length)
+      let animationOptions = UIViewAnimationOptions(rawValue: UInt(animationCurve.rawValue) << 16)
+      
+      var contentInsets = scrollView.contentInset
+      var scrollIndicatorInsets = scrollView.scrollIndicatorInsets
+      
+      contentInsets.bottom = bottomInset
+      scrollIndicatorInsets.bottom = bottomInset
+      
+      UIView.animateWithDuration(animationDuration, delay: 0, options: animationOptions, animations: {
+        scrollView.contentInset = contentInsets
+        scrollView.scrollIndicatorInsets = scrollIndicatorInsets
+        }, completion: nil)
+    }
+  }
+#endif
```

Comment:
I mean we can discuss that tomorrow as I haven't really been part of all discussion before and maybe that's how things were decided.